cni/cni.go

```diff
@@ -192,7 +193,13 @@ func (config *NetworkConfig) Serialize() []byte {
 }
 
 // GetNetworkInfo from the NetworkConfig
-func (config *NetworkConfig) GetNetworkInfo(podNamespace string) *network.NetworkInfo {
+func (config *NetworkConfig) GetNetworkInfo(podNamespace string) (ninfo *network.NetworkInfo, err error) {
+	if config.OptionalFlags.LoopbackDSR {
+		if err := hcn.DSRSupported(); err != nil {
```

Comment:
(I am not well versed in golang, so bear with me) Why is err := hcn.DSRSupported(); included in the if condition? does this if condition basically come down to err != nil and have you tested that?

Reply:
Yes, I actually encountered this condition on RS5 by using older hcsshim version that did not include https://github.com/microsoft/hcsshim/pull/848 (it was using the old DSR version check, hence it failed on RS5).

This inline assignment and error check in the If statement is a common idiom in Golang, e.g. see: https://golang.org/doc/effective_go#if

Reply:
awesome, thanks David! maybe I will use this in the future :)